Integration/AndroidAAR/BabylonNative/src/main/cpp/BabylonNativeJNI.cpp

```diff
@@ -0,0 +1,170 @@
+#include <android/log.h>
+#include <Babylon/Console.h>
+#include <Babylon/RuntimeAndroid.h>
+#include <android/asset_manager.h>
+#include <android/asset_manager_jni.h>
+#include <InputManager.h>
+
+extern "C" {
+JNIEXPORT void JNICALL Java_BabylonNative_Wrapper_initEngine(JNIEnv* env, jobject obj, jobject assetMgr);
+JNIEXPORT void JNICALL Java_BabylonNative_Wrapper_finishEngine(JNIEnv* env, jobject obj);
+JNIEXPORT void JNICALL Java_BabylonNative_Wrapper_surfaceCreated(JNIEnv* env, jobject obj, jobject surface);
+JNIEXPORT void JNICALL Java_BabylonNative_Wrapper_activityOnPause(JNIEnv* env);
+JNIEXPORT void JNICALL Java_BabylonNative_Wrapper_activityOnResume(JNIEnv* env);
+JNIEXPORT void JNICALL Java_BabylonNative_Wrapper_surfaceChanged(JNIEnv* env, jobject obj, jint width, jint height, jobject surface);
+JNIEXPORT void JNICALL Java_BabylonNative_Wrapper_loadScript(JNIEnv* env, jobject obj, jstring path);
+JNIEXPORT void JNICALL Java_BabylonNative_Wrapper_eval(JNIEnv* env, jobject obj, jstring source, jstring sourceURL);
+JNIEXPORT void JNICALL Java_BabylonNative_Wrapper_setTouchInfo(JNIEnv* env, jobject obj, jfloat dx, jfloat dy, jboolean down);
+};
+
+std::unique_ptr<Babylon::RuntimeAndroid> runtime{};
+std::unique_ptr<InputManager::InputBuffer> inputBuffer{};
+
+static AAssetManager* g_assetMgrNative = nullptr;
+static const char* Root = "file://";
+namespace
+{
+    // this is the way to load apk embedded assets.
+    static std::vector<char> GetAssetContents(const char* filename)
+    {
+        std::string filenameStr{filename};
+        if (filenameStr.substr(0, strlen(Root)) == std::string(Root))
+        {
+            filename += strlen(Root) + 1; // + "/"
+        }
+        std::vector<char> buffer;
+        AAsset *asset = AAssetManager_open(g_assetMgrNative, filename,
+                                           AASSET_MODE_UNKNOWN);
+        if (asset != nullptr)
+        {
+            size_t size = AAsset_getLength64(asset);
+            buffer.resize(size);
+            AAsset_read(asset, buffer.data(), size);
+            AAsset_close(asset);
+        }
+        return buffer;
+    }
+}
+
+JNIEXPORT void JNICALL
+Java_BabylonNative_Wrapper_initEngine(JNIEnv* env, jobject obj,
+                                                   jobject assetMgr)
+{
+    auto asset_manager = AAssetManager_fromJava(env, assetMgr);
+    g_assetMgrNative = asset_manager;
+    jboolean iscopy;
+}
+
+JNIEXPORT void JNICALL
+Java_BabylonNative_Wrapper_finishEngine(JNIEnv* env, jobject obj)
+{
+}
+
+JNIEXPORT void JNICALL
+Java_BabylonNative_Wrapper_surfaceCreated(JNIEnv* env, jobject obj, jobject surface)
+{
+    if (!runtime)
+    {
+        ANativeWindow *window = ANativeWindow_fromSurface(env, surface);
+
+        int32_t width  = ANativeWindow_getWidth(window);
+        int32_t height = ANativeWindow_getHeight(window);
+
+        runtime = std::make_unique<Babylon::RuntimeAndroid>(window, Root, width, height);
```

Comment:
I guess for now the runtime's lifetime is tied to the view's lifetime, and we'll break this out later when we support multi-view? Is that tracked in an issue?

Reply:
Can you please open an issue with all your needs on views.

Reply:
https://github.com/BabylonJS/BabylonNative/issues/147